packages/11ty/_plugins/figures/annotation-factory.js

```diff
@@ -0,0 +1,58 @@
+const chalkFactory = require('~lib/chalk')
+const path = require('path')
+const titleCase = require('~plugins/filters/titleCase')
+const { error } = chalkFactory('Figure Processing:IIIF:Annotations')
+
+/**
+ * Quire Annotation
+ */
+module.exports = class AnnotationFactory {
+  constructor(iiifConfig) {
+    this.iiifConfig = iiifConfig
+  }
+
+  get filepath() {
+    return this.figure.preset === "zoom"
+      ? [
+          this.iiifConfig.outputDir,
+          this.data.id,
+          this.iiifConfig.imageServiceDirectory,
+        ].join('/')
+      : [this.iiifConfig.inputDir, this.data.src].join('/');
+  }
+
+  get id() {
+    const { id, label, src } = this.data
+    switch(true) {
+      case !!id:
+        return id
+      case !!label:
+        return label.split(' ').join('-').toLowerCase()
+        break;
+      case !!src:
+        return path.parse(src).name
+        break;
+      default:
+        error(`Error setting ID for annotation on figure "${this.figure.id}". Annotations must have a 'label' or 'src' property.`)
```

Comment:
We might want to use the logger or throw to handle adding the "Error" prefix to logging, thoughts?
```suggestion
       error(`Unable to set an id for annotation on figure "${this.figure.id}". Annotations must have an 'id', 'label', or 'src' property.`)
```

Reply:
The error message is already red, that might be enough?